lib/gateway/Shard.js

```diff
@@ -2069,6 +2069,204 @@ class Shard extends EventEmitter {
                 this.client.userGuildSettings[packet.d.guild_id] = packet.d;
                 break;
             }
+            case "THREAD_CREATE": {
+                const channel = Channel.from(packet.d, this.client);
+                if(!channel.guild) {
+                    channel.guild = this.client.guilds.get(packet.d.guild_id);
+                    if(!channel.guild) {
+                        this.emit("debug", `Received THREAD_CREATE for channel in missing guild ${packet.d.guild_id}`);
+                        break;
+                    }
+                }
+                channel.guild.threads.add(channel, this.client);
+                this.client.threadGuildMap[packet.d.id] = packet.d.guild_id;
+                /**
+                * Fired when a channel is created
+                * @event Client#threadCreate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The channel
+                */
+                this.emit("threadCreate", channel);
+                break;
+            }
+            case "THREAD_UPDATE": {
+                const channel = this.client.getChannel(packet.d.id);
+                if(!channel) {
+                    break;
+                }
```

Comment:
Archived threads on startup that are then un-archived aren't announced with a THREAD_CREATE, only a THREAD_UPDATE - should probably cache those here